src/internal/error/errors.go

```diff
@@ -92,6 +92,8 @@ const (
 	GeneralCode = "UNKNOWN"
 	// DENIED it's used by middleware(readonly, vul and content trust) and returned to docker client to index the request is denied.
 	DENIED = "DENIED"
+	// PROJECTPOLICYVIOLATION ...
+	PROJECTPOLICYVIOLATION = "PROJECTPOLICYVIOLATION"
```

Comment:
ProjectPolicyViolation = "PROJECT_POLICY_VIOLATION"

Reply:
The same to the `DENIED`, it should be `DeniedCode`

Reply:
And add them to the HTTP status code map https://github.com/goharbor/harbor/blob/master/src/server/error/error.go#L27

Reply:
As the error code and HTTP status code are defined in two files, seems we should take the suggestion of @steven-zou that introduce a `Descriptor` for each common error, but this is out of the scope of this PR

Reply:
the code here is just an specification(not a http code) for docker client. No need to add it.

Reply:
and I will enhance error handling as issue mentioned. https://github.com/goharbor/harbor/issues/10553